tasks/lib/generateHTMLReport.js

```diff
@@ -33,16 +33,24 @@ module.exports = function generateHTMLReport(errorFileObj, options, errorFileCou
             dateFormat;
 
         timePortion = timePortion.substr(0, timePortion.lastIndexOf(":")).replace(/:/g, "-");
-        
+
         dateFormat = datePortion + "-" + timePortion;
         folderPath = (options.useTimeStamp === true) ? "w3cErrors-"+ dateFormat : "w3cErrors";
     }
-    var filePathTemp = curruntErrorFile["filename"].split("/"),
-        filePath;
 
-    filePathTemp = (filePathTemp[filePathTemp.length-1].indexOf(".") === -1) ? filePathTemp.slice(filePathTemp.length-2).join("") : filePathTemp.slice(filePathTemp.length-2).join("").split(".")[0];
+    var filePath;
 
-    filePath = filePathTemp + "_validation-report" + ".html";
+    if (!options.errorFileName) {
```

Comment:
Can you please make sure errorFileName should not be blank string and also remove special characters from filename so that it doesn't break while writing the file on system? if you don;t get time then let me know I will do this this week.